src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VideoFanTakeover.java

```diff
@@ -61,29 +66,92 @@ private void runInAdFrame(Lambda f) {
     driver.switchTo().defaultContent();
   }
 
-  public void waitForVideoPlayerVisible() {
-    wait.forElementVisible(UIElementsSelector);
+  public void waitForVideoPlayerVisible(String slotName) {
+    wait.forElementVisible(By.cssSelector(String.format(UI_ELEMENT_SELECTOR, slotName)));
   }
 
-  public void waitForVideoPlayerHidden() {
-    wait.forElementNotVisible(UIElementsSelector);
+  public void waitForVideoPlayerHidden(String slotName) {
+    wait.forElementNotVisible(By.cssSelector(String.format(UI_ELEMENT_SELECTOR, slotName)));
   }
 
-  public void waitForVideoEnd() { wait.forElementPresent(videoContainerSelector, VIDEO_LENGTH); }
+  public void waitforAdToLoad() {
+    runInAdFrame(() -> wait.forElementClickable(playTriggerButtonSelector));
+  }
 
-  public void waitForVideoStart() { waitForVideoPlayerVisible(); }
+  public void waitForVideoEnd(String slotName) {
+    wait.forElementPresent(By.cssSelector(String.format(VIDEO_CONTAINER_SELECTOR, slotName)), VIDEO_LENGTH);
+  }
 
-  public Double getCurrentVideoTime() {
+  public void clickOnAdImage(){
+    runInAdFrame(() -> {
+    Actions action = new Actions(driver);
+    action.moveToElement(wait.forElementClickable(playTriggerButtonSelector), -20, -20).click().build().perform();
```

Comment:
Let's document somehow or ignore and just click the button? :japanese_ogre: I see three options:
* just click the button and ignore the flexible margins around,
* leave a comment describing that the author chose 20px to make sure clicking outside the button works,
* move `-20` and assign it to a descriptive constant.